src/tableOfContentsProvider.ts

```diff
@@ -45,14 +45,20 @@ export class TableOfContentsProvider {
 
     const toc = asciidocDocument
       .findBy({ context: 'section' })
-      .map((section) => ({
-        slug: new Slug(section.getId()),
-        text: section.getTitle(),
-        level: section.getLevel(),
-        line: section.getLineNumber() - 1,
-        location: new vscode.Location(textDocument.uri,
-          new vscode.Position(section.getLineNumber() - 1, 1)),
-      }))
+      .map((section) => {
+        let lineNumber = section.getLineNumber() // Asciidoctor is 1-based but can return 0 (probably a bug/limitation)
```

Comment:
```suggestion
        // Asciidoctor is 1-based but since it has to look for setext section titles by default, it advances from the document title line to the include file. However, when it tries to back up, it ends up at line 0 of that include file instead of coming back out of it. That's how we get to line 0.
        // One way to fix the problem is to turn off setext section titles.
        // Asciidoctor::Compliance.underline_style_section_titles = false
        // Unless someone sees a way to fix this particular otherwise, it probably won't be resolved until Asciidoctor gets reworked to use the parsing grammar being proposed in the specification.
        let lineNumber = section.getLineNumber()
```